### DISCUSSION TITLE
query().limit()....exec() returning any?

### DISCUSSION THREAD

**Discussion by BrianArch96** [dynamoose/dynamoose#1447] [Q&A]
Is there a way we could type this with some generic passed through to the documentReceiver? I see that the return actually does have a structure with an array of items and some other useful info, why is this then always resolved to any?

Also, this limit limits the number of items scanned for a given query, is there a way to limit the items fetched? Similar to the aws-data-mapper [limit](https://github.com/awslabs/dynamodb-data-mapper-js/blob/master/packages/dynamodb-data-mapper/src/namedParameters/QueryOptions.ts#L30)

**Comment by ptejada** [comment]
To get the type hint for `exec()` you must define your model with a type. The documentation for `Model` has an example of how to do this
https://dynamoosejs.com/guide/Model

Here is an example of the type hint from the IDE when a model is defined with a class as a structure:
![image](https://user-images.githubusercontent.com/253405/186332721-aa604980-d316-4c48-a007-33c73f928072.png)

I can't comment about the limit option.
